File: ImageBroadcastApi/Models/ImageModel.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace ImageBroadcastApi.Models
{
    public class ImageModel
    {
        public int Id { get; set; }

        public string FileName { get; set; } = string.Empty;
        public string FilePath { get; set; } = string.Empty;
        public DateTime UploadedAt { get; set; } = DateTime.Now; // 
        // Foreign Key
        public int UserId { get; set; } // NOT NULL
        public User User { get; set; } = null!;
        public bool IsBroadcasted { get; set; } = false;



    }
}

File: image_broadcast_app/lib/models/image_model.dart
class ImageModel {
  final int id;
  final String fileName;
  final String filePath;
  final bool isBroadcasted;

  ImageModel({
    required this.id,
    required this.fileName,
    required this.filePath,
    required this.isBroadcasted,
  });

  factory ImageModel.fromJson(Map<String, dynamic> json) {
    return ImageModel(
      id: json['id'],
      fileName: json['fileName'],
      filePath: json['filePath'],
      isBroadcasted: json['isBroadcasted'] ?? false,
    );
  }
}
